public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderContainer.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useReducer } from 'react';
+import { LokiDatasource } from '../../datasource';
+import { LokiQuery } from '../../types';
+import { buildVisualQueryFromString } from '../parsing';
+import { lokiQueryModeller } from '../LokiQueryModeller';
+import { LokiQueryBuilder } from './LokiQueryBuilder';
+import { QueryPreview } from './QueryPreview';
+import { LokiVisualQuery } from '../types';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+export interface Props {
+  query: LokiQuery;
+  datasource: LokiDatasource;
+  onChange: (update: LokiQuery) => void;
+  onRunQuery: () => void;
+}
+
+export interface State {
+  visQuery?: LokiVisualQuery;
+  expr: string;
+}
+
+/**
+ * This component is here just to contain the translation logic between string query and the visual query builder model.
+ */
+export function LokiQueryBuilderContainer(props: Props) {
+  const { query, onChange, onRunQuery, datasource } = props;
+  const [state, dispatch] = useReducer(stateSlice.reducer, {
+    expr: '',
+    visQuery: {
+      labels: [],
+      operations: [{ id: '__line_contains', params: [''] }],
+    },
+  });
```

Comment:
```suggestion
    const [state, dispatch] = useReducer(stateSlice.reducer, { expr: query.expr });
```

Think this should also work and is simpler.

Reply:
But then we don't get `__line_contains` operation in query builder mode. I liked this solution, because it shows `__line_contains` operation, but it doesn't show it in query text if you don't fill it out. Which is how it is on main. Also if you switch from Builder -> Code, it doesn't add  `{} | ""` into text editor.

Reply:
Here is the video of what do I mean by it: 

https://user-images.githubusercontent.com/30407135/163347292-0c7f98fc-5952-41b1-acd6-326591ca0314.mov